Nested query filter support is now broken

I have recently updated my project to use `react-admin` and `ra-data-feathers`.
After the update I have found that the nested query filters are not working any more.
I have found the following after investigation.

As an example consider react admin is sending this request to `ra-data-feathers`,

```
restClient(GET_LIST, 'users', {
    filter: { createdAt: { $gte: "2018-12-24T14:56:14.274Z" } },
    sort: { field: 'createdAt', order: 'DESC' },
    pagination: { page: 1, perPage: 100 },
})
```

ra-data-feathers would receive this filter,

```
filter: {
    createdAt: {
        $gte: "2018-12-24T14:56:14.274Z"
    }
}
```
This is the format expected by the feathers-client library. Earlier `ra-data-feathers` versions would simply sent this to `feathers-client`, and it would encode it correctly as `createdAt[$gte]: 2018-12-24T15:24:05.521Z` before sending it to `feathers-server`.

However, this line of code

```
Object.assign(query, fetchUtils.flattenObject(params.filter));
```

at https://github.com/josx/ra-data-feathers/blob/master/src/restClient.js#L53 flattens the filter object to,
```
createdAt.$gte: "2018-12-24T14:56:14.274Z"
```
and sends this to `feathers-client`. Since this is not the filter format expected by the `feathers-client` it sends it as it is to the feathers-server. As a result server issues a db query with a` WHERE user.createdAt.$gte="2018-12-24T14:56:14.274Z"` to the database and the database throws a column does not exist error.

I see that the problematic line (ie, flattening of filters) was added in #53. I am not sure about's it's objective. If the flattening was intended for some client library other than `feathers-client`, then I believe the flattening should be done within that client library. Hence I feel the fix would be to simply revert it back to

`Object.assign(query, params.filter);`
 
I can send a PR for the same.
@josx What do you think?

Hey josx,

> Tests are not passing, because of your change (because doesnt support deep props)

Are you sure about this? I've just run the test suite at the current commit (104260ea1077d0f6fdd5761f1871e2b1f82b7d5f) and it appears there is a test failing in the latest master commit at least.

`  1) Rest Client
       when called with GET_LIST
         formats params into a query and pass it to client:
     AssertionError: expected false to be true`

When I add in the changes from #91 , I get the same failed test + no new failing tests.

--- a/src/restClient.js
+++ b/src/restClient.js
@@ -50,7 +50,7 @@ export default (client, options = {}) => {
             [field === 'id' ? idKey : field]: order === 'DESC' ? -1 : 1,
           };
         }
-        Object.assign(query, fetchUtils.flattenObject(params.filter));
+        Object.assign(query, params.filter);
         dbg('query=%o', query);
         return service.find({ query });
       case GET_ONE:

--- a/test/restClient.spec.js
+++ b/test/restClient.spec.js
@@ -186,7 +186,9 @@ describe('Rest Client', function () {
           id: -1,
         },
         name: 'john',
-        'address.city': 'London',
+        address: {
+          city: 'London',
+        }
       };
       return asyncResult.then(() => {
         expect(fakeService.find.calledWith({

